File: randomProblems/twoSum/twoSum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        valDict = { target-nums[n]:n for n in range(len(nums))}
        for index in range(len(nums)):
            targetIndex = valDict[target - nums[index]]
            if targetIndex != index:
                return [index, targetIndex]
            
        return 0
        pass 
    
s = Solution()

nums = [2,7,11,15]
target = 9
print(s.twoSum(nums, target))

File: practice-problems/twosum-1/twosum.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> indexDict = new HashMap<>();

        for(int i = 0; i<nums.length; i++){
            indexDict.put(nums[i], i);
        }

        for(int j = 0; j< nums.length; j++ ){
            int curr = nums[j];
            int needed = target - curr;
            if( indexDict.containsKey(needed) && indexDict.get(needed) != j){
                return new int[] {j, indexDict.get(needed)};
            }
        }
        return new int[] {-1,-1};
    }
}